rnote-ui/src/overlays.rs

```diff
@@ -260,7 +262,34 @@ impl RnOverlays {
                                 appwindow.handle_widget_flags(widget_flags, &canvas);
                             }
                         }
-                        PenStyle::Brush | PenStyle::Shaper | PenStyle::Eraser | PenStyle::Selector | PenStyle::Tools => {}
+                        PenStyle::Selector => {
+                            let mut widget_flags = engine.record(Instant::now());
```

Comment:
I'd move all of this into the function in the engine, and let it return `WidgetFlags` with `redraw` and `store_modified` flagged true. Then the code in the UI is reduced to two lines: the function call and `appwindow.handle_widget_flags()`

Reply:
I meant as a method on the store. 

> then it would need an engine view (specifically the task sender and camera) as a parameter
thats true. Is it a huge issue to have that as "public" API from the perspective of rnote-ui ? I think typewriter does that as well. Otherwise there are some functions where I simply added a doc comment with "strokes then need to update their rendering", which also has the advantage avoiding iterating over the currently viewed strokes multiple times when those modifications are chained (not in this case, but for example, selection-duplication).
I'd then follow (as you said) it by a call to `canvas.update_rendering()`

Reply:
I eventually want to refactor this part by some sort of automatic diffing between the current and previous state of the engine after making modifications. We already have the immutable,  cheap-to-clone & clone-on-write (the `Arc`'s) state to do that

Reply:
> Is it an issue to have that as "public" API from the perspective of rnote-ui ? I think typewriter does that as well.

Probably not, and yeah, typewriter does it too. Actually, I'd only pass the task sender and camera as parameters because `RnoteEngine#store` is already being borrowed while calling the `change_fill/stroke_color` function. Or we go for the alternative you described, but that would (at least ATM) needlessly re-render the background.

Reply:
It would only reapply the prerendered background tile onto the gsk scene graph :) there is a comment that says creating the texture is "expensive" but that is only in the context of that function.

I think it would be best to keep the rerendering out of these functions. So that the flow is `strokes are modified -> rendering flagged dirty -> finish up with rerendering those that are dirty`. This is consistent with other functions, and there is also the benefit that if we want to reuse it somewhere, we don't end up having to thread `EngineView` or `Camera / tasks_tx` through them.

Reply:
Alright. I've used `engine.update_rendering_current_viewport()` though, because engine is already borrowed and the canvas function does more unneeded stuff :P

Reply:
I just added [update_content_rendering_current_viewport(&mut self))[https://github.com/flxzt/rnote/blob/9637527bd86d28441a369e926d001b50c0ef5643/rnote-engine/src/engine/rendering.rs#L46] for this